### PR TITLE
Support to regular expression mux

### PR DESCRIPTION
**Pull request by longdeqidao**

### CODE REVIEW

File: plugins/plugins.go [hunk 1]
```diff
@@ -52,6 +53,37 @@ type HTTPMux interface {
...
+	Pattern       *HTTPRegExpPattern
+	RE            *regexp.Regexp
+	LiteralRE     string // redundant but for introspection
+	LiteralPrefix string
+	Entries       map[string]*HTTPMuxEntry
+}
+
+type PipelineRTable map[string]*MethodsEntry
+
+type HTTPRegExpMux interface {
```
Comment (LiuZhiyan) [inline plugins/plugins.go]: it seems this might take the place of HTTPMux stuff? (drop HTTPMux and rename this)

Comment (longdeqidao) [reply plugins/plugins.go]: Sure. (My original idea is just to keep HTTPMux implementation in EG repo which you can see it in another mux PR, so is that OK to delete all original mux implementation?)

Comment (LiuZhiyan) [reply plugins/plugins.go]: The stuff in my mind is:

```

                    +--------------------------------------------------------+
                    |                                                        |
                    |    unify (golang) interface as standard http router    |
                    |                                                        |
                    +--------------------------------------------------------+
                          |                      |                     |
                          |                      |                     |
                          |                      |                     |
                          |                      |                     |
   +----------------------v---------------+      |  +------------------v---------------------+
   |                                      |      |  |                                        |
   |  1. original router implementation   |      |  |  2. new regex router implementation    |
   |  (no regex support, http_mux.go)     |      |  |   (http_regex_mux.go, something like that)
   |                                      |      |  |                                        |
   +--------------------------------------+      |  +----------------------------------------+
                                                 |
                                           +-----v--------------------------------+
                                           |                                      |
                                           |  3. other routers,                   |
                                           |  could be added in future.           |
                                           |                                      |
                                           +--------------------------------------+


```

Comment (longdeqidao) [reply plugins/plugins.go]: Got it. Cool, this makes sense to me. Will do the adaption for the original router implementation and adjust interface more common after test functionality of new implementation.

**Comment by longdeqidao** [comment]
This `add-regexp-mux` branch is used in iget, can't push -f it. close and open a new one: #5 .

**Comment by longdeqidao** [comment]
Replaced by #5

**Pull request closed without merging**